src/classes/form-actions.ts

```diff
@@ -0,0 +1,137 @@
+import uuid from 'uuid'
+import { FormModel } from 'src/solid-forms'
+
+export class FormActions {
+  constructor(private formModel: any, private formObject: any) {}
+
+  static getSubjectLinkId = (currentLink: string) => {
+    if (currentLink.includes('#')) {
+      const id = Date.now()
+      return `${currentLink.split('#')[0]}#${id}`
+    }
+  }
+
+  static cleanFieldNode = (field: any) => {
+    let updatedField = field
+
+    if (updatedField && updatedField['ui:parts']) {
+      for (const childKey in updatedField['ui:parts']) {
+        updatedField = {
+          ...updatedField,
+          'ui:parts': {
```

Comment:
Put this string in a const since we use it a few times, maybe the rest of the predicates strings as well

Reply:
done